desc/magnetic_fields/_current_potential.py

```diff
@@ -842,3 +1042,790 @@ def nfp_loop(j, f):
+    # from the indices, calculate the actual zeta and theta values of the contours
+    contours_theta_zeta_not_uniform = [
+        np.array(
+            [
+                np.interp(contour[:, 0], np.arange(zeta_full.size), zeta_full),
+                np.interp(contour[:, 1], np.arange(theta_full.size), theta_full),
+            ]
+        ).T
+        for contour in contours_indices
+    ]
+    # make all the contours the same length by interpolating
+    # any shorter ones to match the length of the longest
+    Npts = np.max([c.shape[0] for c in contours_indices])
+    s = np.linspace(0, 2 * np.pi, Npts)
+    contours_theta_zeta = [
+        np.array(
+            [
+                np.interp(s, np.linspace(0, 2 * np.pi, c[:, 0].size), c[:, 0]),
+                np.interp(s, np.linspace(0, 2 * np.pi, c[:, 1].size), c[:, 1]),
+            ]
+        ).T
+        for c in contours_theta_zeta_not_uniform
+    ]
+    # reverse so that our coils start at zeta=0, this is just a choice
+    contours_theta_zeta.reverse()
+
+    numCoils = 0
+    # to be used to check closure conditions on the coils
+    ## closure condition in zeta for modular is returns to same zeta,
+    ## while for helical is that the contour dzeta = 2pi/NFP
+    zeta_diff = 2 * jnp.pi / nfp if not jnp.isclose(helicity, 0) else 0.0
+    ## closure condition in theta for modular is that dtheta = 2pi,
+    ## while for helical the dtheta = 2pi*abs(helicity)
+    theta_diff = (
+        2 * jnp.pi * jnp.abs(helicity) if not jnp.isclose(helicity, 0) else 2 * jnp.pi
+    )
+
+    if show_plots:
+        plt.contourf(
+            zeta_full_2D.T, theta_full_2D.T, jnp.transpose(phi_total_full), contours
+        )
+        plt.xlabel(r"$\zeta$")
+        plt.ylabel(r"$\theta$")
+    for j in range(N_trial_contours):
+        contour_zeta.append(contours_theta_zeta[j][:, 0])
+        contour_theta.append(contours_theta_zeta[j][:, 1])
+        # check if closed and if not throw warning
+
+        if not jnp.isclose(
+            jnp.abs(contour_zeta[-1][-1] - contour_zeta[-1][0]), zeta_diff
+        ) or not jnp.isclose(
+            jnp.abs(contour_theta[-1][-1] - contour_theta[-1][0]), theta_diff
+        ):
+            warnings.warn(
+                f"Detected a coil contour (coil index {j}) that may not be "
+                "closed, this may lead to incorrect coils, "
+                "check that the surface current potential contours do not contain "
+                "any local maxima or window-pane-like structures,"
+                " and that the current potential contours do not go across "
+                "The edges of the zeta extent used for the plotting:"
+                "the zeta=0 or zeta=2pi/NFP planes for helical coils or the"
+                " zeta=-pi/NFP and zeta=2pi+pi/NFP planes, for modular coils. "
+                "Use `show_plots=True` to visualize the contours.",
+                UserWarning,
+            )
+        numCoils += 1
+        if show_plots:
```

Comment:
plotting the contours themselves over the contourf plot is useful to see the actual coil contour curves, instead of just looking at the filled contour plot